python/lib/utilities.py

```diff
@@ -173,22 +173,25 @@ def update_set_file_path_info(set_file, fdt_file):
     # grep the basename without the extension of set_file
     basename = os.path.splitext(os.path.basename(set_file))[0]
 
-    # read the .set EEG file using scipy
-    dataset = scipy.io.loadmat(set_file)
-
-    # update the EEG paths in the .set file
-    if 'filename' in dataset.keys():
-        dataset['filename'] = numpy.array(basename + ".set")
-    if 'setname' in dataset.keys():
-        dataset['setname'] = numpy.array(basename)
-    if 'EEG' in dataset.keys():
-        dataset['EEG'][0][0][1] = numpy.array(basename + ".set")
-    if fdt_file and 'EEG' in dataset.keys():
-        dataset['EEG'][0][0][15] = numpy.array(basename + ".fdt")
-        dataset['EEG'][0][0][40] = numpy.array(basename + ".fdt")
-
-    # write the new .set file with the correct path info
-    scipy.io.savemat(set_file, dataset, False)
+    try:
+        # read the .set EEG file using scipy
+        dataset = scipy.io.loadmat(set_file)
+
+        # update the EEG paths in the .set file
+        if 'filename' in dataset.keys():
+            dataset['filename'] = numpy.array(basename + ".set")
+        if 'setname' in dataset.keys():
+            dataset['setname'] = numpy.array(basename)
+        if 'EEG' in dataset.keys():
+            dataset['EEG'][0][0][1] = numpy.array(basename + ".set")
+        if fdt_file and 'EEG' in dataset.keys():
+            dataset['EEG'][0][0][15] = numpy.array(basename + ".fdt")
+            dataset['EEG'][0][0][40] = numpy.array(basename + ".fdt")
+
+        # write the new .set file with the correct path info
+        scipy.io.savemat(set_file, dataset, False)
+    except NotImplementedError:     # Thrown for matlab v7.3 files
+        pass    # Assumes path fields are already set
```

Comment:
@jeffersoncasimir Thank you for the PR!!! Would it be worth to have a warning printed here mentioning that the pipeline assumes that the path fields are already set?

Reply:
@cmadjar Yes, but upon reconsideration, I think I should change it so that we aren't assuming that the fields are properly set, and we should instead use a library that can read the fields and only allow it on that condition. This could otherwise potentially introduce a vector for referencing unauthorized data.

Reply:
Thanks, I'll add to the Loris-EEG agenda for this Friday - Jefferson could you bring an estimate of how long will this approach take?